### ISSUE TITLE
Multirotor Mission Corner Cases

### ISSUE THREAD

**Issue by tumbili** [PX4/Firmware#3516]
This is a list of corner cases / problems:

 - [x] The navigator uses a certain default acceptance radius for takeoff. If now your takeoff waypoint altitude is equal or lower to this you can imagine what happens
Implemented in #3518 
 - [ ] acceptance radius handling: (issue #3566)
 - [x] we should also be limiting vertical accleration. E.g. after takeoff if the vehicle is still supposed to increase altitude the thrust setpoint can just do a huge jump
Implemented in #3524 
 - [x] need to revisit landing logic. In the simulation the plane drops to the ground at one point. The reason it because the controller was not allowed to increase it's thrust anymore here https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L1497
If I comment out this part everything works fine
Fix proposed in #3526
 - [x] during the landing the plane does not seem to care much about it's horizontal position anymore
 - [x] Landing waypoint. Depending on the location of the landing waypoint the vehicle might reach the ground before it has reached the waypoint. It should first reach the location and then start to descend.
Implemented in #3523 
 - [x] I can't see any effect of the landing waypoint altitude on the behaviour of the plane
Fixed in #3523
 - [ ] There must be a special handling of the yaw setpoint for missions. During a backtransition the multirotor code tries to achieve the yaw setpoint specified in the mission item. This can lead to a 'bent' transition path. For manual mode the yaw setpoint is reset in the mc position controller. (issue #3564)

- [ ] MC: Landing ground detection needs improvements #3560 
- [ ] MC: if takeoff speed is higher than max vel z it can overshoot in height #3561 
- [ ] reset takeoff states in position controller on arming to prevent throttle ramp being finished when switching to mission before arming (https://github.com/PX4/Firmware/pull/3605, https://github.com/PX4/Firmware/issues/3608)

**Comment by ecmnet** [comment]
Comment from my side: It might be a good idea to separate the code of state control, setpoint calculation and target-attitude calculation a bit more. At least for me it was quite hard to follow the different modes of operation within mc_pos_control_main.cpp. This would also reduce the risk of side effects...

**Comment by tumbili** [comment]
@ecmnet Generally I agree with you. However, if you want to do this you will have to test everything through again, which needs time.

**Comment by ecmnet** [comment]
Yes, you're right. We will see, whether the list of issues grow. If it does and it is faster than resolving them, it would be time for refactoring.

**Comment by edigotlieb** [comment]
happened to me too - caused a deadly crash because i set the takeoff altitude to 10m

**Comment by LorenzMeier** [comment]
@tumbili Not sure we should add an acceptance radius in QGC for takeoff. Now that I'm thinking about it I even think the Firmware should always use its default. That will always yield a good value for all platform types and depend properly on the flight mode for VTOL. These are all things that QGC doesn't know about. So my proposal would be to ignore the acceptance radius completely, in all cases.

**Comment by tumbili** [comment]
Probably everything fixed, need revaluation anyway.
